src/gobimportclient/validator/__init__.py

```diff
@@ -8,6 +8,17 @@
 """
 import re
 
+from gobcore.exceptions import GOBException
+
+
+def _validate_primary_key(entities, entity_id):
+    primary_keys = set()
+    for entity in entities:
```

Comment:
Can't you just transform the entities id's into a set and compare the size?
Or else, report all duplicate keys instead of only the first duplicate found

Reply:
I've added reporting on all duplicate keys